soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -3947,14 +3945,18 @@ void DrawRandoEditor(bool& open) {
                 ImGui::PushItemWidth(-FLT_MIN);
 
                 //Starting Age
-                //Disabled when Forest is set to Closed
-                bool disableRandoStartingAge = !CVar_GetS32("gRandomizeForest", 0);
-                const char* disableRandoStartingAgeText = "This option is disabled because \"Forest\" is set to \"Closed\".";
+                //Disabled when Forest is set to Closed or under very specific conditions
+                //RANDOTODO: Replace magic number checks with enums
+                bool disableRandoStartingAge = (CVar_GetS32("gRandomizeForest", 0) == 0) || //If Forest is closed...
+                    (CVar_GetS32("gRandomizeDoorOfTime", 0) == 0) && //or Door of Time is Closed
+                    (CVar_GetS32("gRandomizeShuffleOcarinas", 0) == 0) && //and ocarinas are not shuffled
+                    (CVar_GetS32("gRandomizeLogicRules", 0) == 0); // and logic is glitchless
+                const char* disableRandoStartingAgeText = "This option is disabled due to other options making the game unbeatable.";
                 ImGui::Text(Settings::StartingAge.GetName().c_str());
                 UIWidgets::InsertHelpHoverText(
                     "Choose which age Link will start as.\n\n"
                     "Starting as adult means you start with the Master Sword in your inventory.\n"
-                    "Only the child option is compatible with Closed Forest."    
+                    "The child option is forcefully set if it would conflict with other options."    
```

Comment:
same as previous comment, it'd be nice to not have it just be a vague "other options" but i'm not sure how to handle that without it getting super wordy